### PR TITLE
SALTO-3108: Support installing Salesforce packages using Salto

### PR DESCRIPTION
**Pull request by tamtamirr**
Fixed a bug that didn't allow users to install packages on their orgs using the Salesforce Adapter.

---

noise suppression PR: https://github.com/salto-io/salto_private/pull/4960

The bug was caused due to the fact the field `activateRSS` on the MetadataType `InstalledPackage` always returns as undefined from Salesforce (speaking of XML presentation: `{ attr_xsi_nil: true }`). According to [this](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_installedpackage.htm) docs, the default value is false so I felt comfortable setting it to false upon fetch for all of the `InstalledPackage` instances.

---
_Release Notes_: 
Salesforce Adapter:
- Fixed a bug that didn't allow installing packages using the adapter's deploy (of `InstalledPackage` instance)

---
_User Notifications_: 
Salesforce:
- The field `activateRSS` will be added to your existing `InstalledPackage` instances.

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/add_default_activate_rss.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+* limitations under the License.
+*/
+import { Element, InstanceElement, isInstanceElement } from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { FilterWith } from '../filter'
+import { isInstanceOfType } from './utils'
+import { INSTALLED_PACKAGE_METADATA } from '../change_validators/package'
+import { ACTIVATE_RSS } from '../constants'
+
+const isInstanceOfTypeInstalledPackage = isInstanceOfType(INSTALLED_PACKAGE_METADATA)
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/add_default_activate_rss.ts:23]: nit: This function seems redundant to me

File: packages/salesforce-adapter/src/filters/add_default_activate_rss.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Element, InstanceElement, isInstanceElement } from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { FilterWith } from '../filter'
+import { isInstanceOfType } from './utils'
+import { INSTALLED_PACKAGE_METADATA } from '../change_validators/package'
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/add_default_activate_rss.ts:20]: Can we move this const to the `constants` file please?

File: packages/salesforce-adapter/src/filters/add_default_activate_rss.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const { awu } = collections.asynciterable
+
+const setDefaultActivateRSSValue = ({ value }: InstanceElement): void => {
+  value[ACTIVATE_RSS] = false
+}
+
+/**
+ * Fixing InstalledPackage instances to the default activateRSS value: false
+ * Salesforce never returns this value, and instead `{ xsi:nil: true }` is returned
```
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/add_default_activate_rss.ts:33]: @adamro { xsi:nil: true } (Also through SFDX)

File: packages/salesforce-adapter/src/filters/add_default_activate_rss.ts (hunk 1 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/add_default_activate_rss.ts:23]: v

File: packages/salesforce-adapter/src/filters/add_default_activate_rss.ts (hunk 2 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/add_default_activate_rss.ts:20]: v

File: packages/salesforce-adapter/src/filters/add_default_activate_rss.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { Element, InstanceElement, isInstanceElement } from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { FilterWith } from '../filter'
+import { isInstanceOfType } from './utils'
+import { INSTALLED_PACKAGE_METADATA } from '../change_validators/package'
+import { ACTIVATE_RSS } from '../constants'
+
+const isInstanceOfTypeInstalledPackage = isInstanceOfType(INSTALLED_PACKAGE_METADATA)
+
+const { awu } = collections.asynciterable
```
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/add_default_activate_rss.ts:25]: v

**Review by adamro — APPROVED** [review]

**Pull request merged**